### DISCUSSION TITLE
Issue with Power Adjustment of Solar Module

### DISCUSSION THREAD

**Discussion by lin-1l1** [CURENT/andes#534] [Q&A] [answered]
Dear Andes developers,

I have reviewed the "Change Setpoint" example and would like to adjust the power output of the solar module in the ieee14_solar.xlsx file in a similar manner. I have attempted to adjust the real power output of the solar energy in the TDS. However, despite consulting the REPC, REEC, and REGC sections of the manual and adjusting some parameters (such as Plant_ref, Pline0, and Ipcmd0), the output power does not seem to change.

In this situation, I would like to inquire about which parameters should be controlled to effectively alter the real power output of the solar energy? Are there any other aspects I should consider or areas I might have overlooked?

Thank you very much for your valuable assistance!

**Comment by cuihantao** [comment]
Thank you for the questions! Changing the reference points for renewable generators are much more involved. I will make you an example in the coming week.

**Reply by lin-1l1** [reply]
Dear Professor Cui
Thank you very much for your response and willingness to provide an example. I truly appreciate your assistance with this matter.

**Comment by cuihantao** [comment, marked as answer]
There isn't an elegant way to change the active power reference. But you can hack it like this:

In `REGCA1`, the active power output `Pe` is computed by `Ipcmd * v`, and `Ipcmd = Ipcmd0_LVG`.

`Ipcmd0_LVG` is a type of constant service. You can impact ' Pe ' by changing the value of `Ipcmd0_LVG.v[:]`. At steady-state, the desired `Pe` can be achieved by setting `Ipcmd0_LVG.v[:] = desired_Pe / voltages`.

**Reply by lin-1l1** [reply]
Thank you, Professor. I would like to express my gratitude once again for providing the example, which has been immensely helpful for my research.

I have a few new questions that I would like to consult you about (regarding ROCOF and inertia constant H, among others). Would you be available to answer them? If possible, could we communicate via email?

Thank you for taking the time to address my questions, and for your valuable time and assistance.

**Reply by cuihantao** [reply]
Thanks! It would be helpful to post it to Discussion for future reference. For anything non-public, feel free to email me!
